synapse/config/_base.py

```diff
@@ -145,6 +145,12 @@ def load_config(cls, description, argv, generate_section=None):
             help="Generate any missing key files then exit"
         )
         config_parser.add_argument(
+            "--generated-directory",
+            metavar="DIRECTORY",
+            help="Used with 'generate-*' options to specify where generated"
+                 " files (such as certs and signing keys) should be stored."
```

Comment:
It might be worth saying that if the location of a key file is explicitly given in the config then that location will be used.